ibm/service/power/data_source_ibm_pi_network_interface.go

```diff
@@ -0,0 +1,156 @@
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_NetworkID: {
+				Description:  "Network ID.",
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_NetworkInterfaceID: {
+				Description:  "Network Interface ID.",
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			// Attributes
+			Attr_CRN: {
+				Computed:    true,
+				Description: "The Network Interface's crn.",
+				Type:        schema.TypeString,
+			},
+			Attr_Instance: {
+				Computed:    true,
+				Description: "The attached instance to this Network Interface.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_Href: {
+							Computed:    true,
+							Description: "Link to instance resource.",
+							Type:        schema.TypeString,
+						},
+						Attr_InstanceID: {
+							Computed:    true,
+							Description: "The attached instance ID.",
+							Type:        schema.TypeString,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+			Attr_IPAddress: {
+				Computed:    true,
+				Description: "The ip address of this Network Interface.",
+				Type:        schema.TypeString,
+			},
+			Attr_MacAddress: {
+				Computed:    true,
+				Description: "The mac address of the Network Interface.",
+				Type:        schema.TypeString,
+			},
+			Attr_Name: {
+				Computed:    true,
+				Description: "Name of the Network Interface (not unique or indexable).",
+				Type:        schema.TypeString,
+			},
+			Attr_NetworkSecurityGroupID: {
+				Computed:    true,
+				Description: "ID of the Network Security Group the network interface will be added to.",
+				Type:        schema.TypeString,
+			},
+			Attr_Status: {
+				Computed:    true,
+				Description: "The status of the network address group.",
+				Type:        schema.TypeString,
+			},
+			Attr_UserTags: {
+				Computed:    true,
+				Description: "List of user tags attached to the resource.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Set:         schema.HashString,
+				Type:        schema.TypeSet,
+			},
+		},
+	}
+}
+
+func dataSourceIBMPINetworkInterfaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	networkID := d.Get(Arg_NetworkID).(string)
+	networkInterfaceID := d.Get(Arg_NetworkInterfaceID).(string)
+	networkC := instance.NewIBMPINetworkClient(ctx, sess, cloudInstanceID)
+	networkInterface, err := networkC.GetNetworkInterface(networkID, networkInterfaceID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(*networkInterface.ID)
```

Comment:
To be on safer side can we use `networkID/networkInterfaceID` here?

Reply:
What do you mean by safer side? 
If you are think of `ID` being nil, there will be an error return since that is required field.

Reply:
What I mean is every networkInterfaceID is defined under networkID right? Similar to the resource id which is `cloud_instance_id/network_id/network_interface_id` the data source id can be `network_id/network_interface_id`